### PR TITLE
Ensure that AdvSimd.Insert doesn't zero out the upper bits

### LINKED ISSUE #106079: JIT: Bad codegen with `AdvSimd.Insert`
```csharp
// Generated by Fuzzlyn v2.2 on 2024-08-06 15:11:52
// Run on Arm64 MacOS
// Seed: 289142602786847481-vectort,vector64,vector128,armadvsimd,armadvsimdarm64,armaes,armarmbase,armarmbasearm64,armcrc32,armcrc32arm64,armdp,armrdm,armrdmarm64,armsha1,armsha256
// Reduced from 87.5 KiB to 0.4 KiB in 00:00:36
// Debug: Outputs <4607182418800017408, 13871573557235963454>
// Release: Outputs <4607182418800017408, 0>
using System;
using System.Runtime.CompilerServices;
using System.Numerics;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.Arm;

public class Program
{
    public static Vector128<double> s_38 = Vector128.Create(0, -567.3319449449843d);
    public static void Main()
    {
        double vr4 = 1;
        s_38 = AdvSimd.Insert(s_38, 0, vr4);
        System.Console.WriteLine(s_38.AsUInt64());
    }
}
```

cc @dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #106079 

This bug was introduced in the first Arm64 support was added back in .NET 5: https://github.com/dotnet/runtime/pull/35030

### CODE REVIEW

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp [hunk 1]
```diff
@@ -1349,44 +1349,38 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
...
                 assert(isRMW);
 
                 GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
 
-                if (intrin.op3->isContainedFltOrDblImmed())
-                {
-                    assert(intrin.op2->isContainedIntOrIImmed());
-                    assert(intrin.op2->AsIntCon()->gtIconVal == 0);
+                // fmov (scalar) zeros the upper bits and is not safe to use
+                assert(!intrin.op3->isContainedFltOrDblImmed());
```
Comment (tannergooding) [inline src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1358]: GitHub is displaying the diff a bit weirdly. This is just removing the `if` and fixing the indentation of the `else` block now that its the only code path.

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib for review, this resolves #106079 and needs backport to .NET 9. It should also be considered for backport to .NET 8

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**